epigraphhub/data/brasil/sinan/loading.py

```diff
@@ -1,55 +1,85 @@
+from . import DISEASES, normalize_str
 
+logger.add(SINAN_LOG_PATH, retention="7 days")
 engine = get_engine(credential_name=env.db.default_credential)
 
 
-def upload():
+def upload(parquet_dirs: list):
     """
-    Connects to the EGH SQL server and load all the chunks for all
-    diseases found at `$PYSUS_DATA_PATH` into database. This method cleans
-    the chunks left.
-
+    Connects to the EpiGraphHub SQL server and load parquet chunks within
+    directories, extracted using `extract.download`, into database. Receives
+    a list of parquets directories with their full path, extract theirs
+    DataFrames and upsert rows to Postgres connection following EGH table
+    convention, see more in EGH's documentation:
+    https://epigraphhub.readthedocs.io/en/latest/instruction_name_tables.html#about-metadata-tables
+    Usage:
+    ```
+    upload([
+        '/tmp/pysus/ZIKABR17.parquet',
+        '/tmp/pysus/ZIKABR18.parquet',
+        '/tmp/pysus/ZIKABR19.parquet',
+    ])
     """
-    diseases_dir = Path(PYSUS_DATA_PATH).glob("*")
-    di_years_dir = [x for x in diseases_dir if x.is_dir()]
+    di_codes = {code: name for name, code in DISEASES.items()}
 
-    for dir in di_years_dir:
+    for dir in parquet_dirs:
         if "parquet" in Path(dir).suffix and any(os.listdir(dir)):
-            df = to_df(str(dir), clean_after_read=False)
-            df.columns = df.columns.str.lower()
-            df.index.name = "index"
 
-            table_i = str(dir).split("/")[-1].split(".parquet")[0]
-            table = table_i[:-4].lower()
+            df = _read_parquets_dir(dir)
+            disease_code = str(dir).split("/")[-1].split(".parquet")[0][:-4]
+            tablename = "sinan_" + normalize_str(di_codes[disease_code]) + "_m"
             schema = "brasil"
 
+            print(f"Inserting {dir} on {schema}.{tablename}")
+
             with engine.connect() as conn:
                 try:
-
                     upsert(
                         con=conn,
                         df=df,
-                        table_name=table,
+                        table_name=tablename,
                         schema=schema,
                         if_row_exists="update",
                         chunksize=1000,
                         add_new_columns=True,
                         create_table=True,
                     )
 
-                    logger.info(f"Table {table} updated")
+                    print(f"Table {tablename} updated")
 
                 except Exception as e:
-                    logger.error(f"Not able to upsert {table} \n{e}")
+                    logger.error(f"Not able to upsert {tablename} \n{e}")
                     raise e
+
+
+def _read_parquets_dir(path: str) -> pd.DataFrame:
+    chunks = Path(path).glob("*.parquet")
+    chunks_dfs = list()
+
+    try:
+        for parquet in chunks:
+            df = pd.read_parquet(str(parquet), engine="fastparquet")
+            objs = df.select_dtypes(object)
```

Comment:
Have you tested this for potentially unwanted side effects?
You are doing a replacement on multiple columns, without checking first if they have the issue of the null character.

I think this would be safer if we performed the replacement in a try/except clause.

Also, we need a test that verifies that, if we run a **well-formed** parquet through this function, it will not be modified in any way.